src/hotspot/share/memory/archiveUtils.hpp

```diff
@@ -30,6 +30,8 @@
 #include "runtime/arguments.hpp"
 #include "utilities/bitMap.hpp"
 
+#define SharedSpaceObjectAlignment      8
```

Comment:
I think we can use a `const int` instead of `#define`. Also, some comments will be helpful. Maybe: 

```
// Metaspace::allocate() requires that all blocks must be aligned with KlassAlignmentInBytes.
// We enforce the same alignment rule in blocks allocated from the shared space.
const int SharedSpaceObjectAlignment = KlassAlignmentInBytes;
```

Reply:
+1. Please do not hide any dependencies on Klass alignment (I am currently experimenting with increasing class space allocation alignment to increase the range of zero-based Klass* encoding)

Reply:
Thanks @iklam and @tstuefe for the review.
I've made the change in webrev 01.